draft-ietf-rats-corim.md

```diff
@@ -1733,19 +1743,35 @@ An ARS is a list of ECTs that describe ACS entries that are selected for use as
 {::include cddl/intrep-ars.cddl}
 ~~~
 
+## Session setup (Phase 0) {#sec-phase0}
+
+The exchange of a request for attestation appraisal for a response of Attestation Results corresponds to a single Attestation Session.
+
+During this setup phase, the Verifier populates its Appraisal Session with a consistent view of all its inputs to the Appraisal Procedure.
+Inputs are various conceptual messages collected from Reference Value Providers, Endorsers, Verifier Owners, and Attesters.
+Conceptual messages may include Attestation Evidence, CoMID tags ({{sec-comid}}), CoSWID tags {{-coswid}}, CoBOM tags ({{sec-cobom}}), and static cryptographic validation key material (including raw public keys, root certificates, intermediate CA certificate chains, and Concise Trust Anchor Stores (CoTS) {{-ta-store}}.
+It is left to Verifier Policy to determine if input sources must use supply chain transparency constructs (see {{-scitt-arch}}) to track input provenance.
+
+The time at which the Verifier evaluates certificate- and tag-validity is an input.
+Once the Appraisal Session inputs are collected, no more may be added to the Appraisal Session apart from one exception.
+Certificate revocation status results may be collected during Phase 1, since there is no collective simultaneity of all responses.
+
+It is left to Verifier Policy to determine if or how to log the inputs used for a given Appraisal Session for optional use in Attestation Results.
+
+Note: Verifiers may rely on conveyance protocol–specific context to identify an Evidence source, which is the Evidence input oracle for appraisal.
+
 ## Input Validation and Transformation (Phase 1) {#sec-phase1}
 
-During the initialization phase, the CoRIM Appraisal Context is loaded with various conceptual message inputs such as CoMID tags ({{sec-comid}}), CoSWID tags {{-coswid}}, CoBOM tags {{-cobom}}, and cryptographic validation key material (including raw public keys, root certificates, intermediate CA certificate chains), and Concise Trust Anchor Stores (CoTS) {{-ta-store}}.
-These objects will be utilized in the Evidence Appraisal phase that follows.
-The primary goal of this phase is to ensure that all necessary information is available for subsequent processing.
+In Phase 1 the Verifier constructs an Appraisal Context that will serve as the set of valid sources of information for the Appraisal Procedure.
+The primary goal of this phase is to ensure that all necessary information is valid and available for subsequent processing.
 
-After context initialization, additional inputs are held back until appraisal processing has completed.
+No inputs other than dynamically-fetched information about certificate revocation status (see {{RFC6960}}, and {{Section 4.2.1.13 of -pkix-cert}}) may be collected in Phase 1.
```

Comment:
I thought the previous text was better / orthogonal to the suggested text.
```suggestion
After the Appraisal Context is initialized, appraisal processing is allowed to complete before new inputs are considered.
```
I thought the previous text was better / orthogonal to the suggested text.

Reply:
The “allowed” worries me, since we don’t require monotonic matching conditions for profiles and mid-appraisal additions of tags can lead to inconsistent ACS results.

Reply:
Party pooper mode on. (For different reasons) I don't like either 🤣

Reply:
I've added a hermeticity subsection to describe this without any need for operational suggestions like having inputs be "held back".